File: apps/workflow-b/src/handlers/manual-review.handler.ts
import type { Context } from 'aws-lambda';
import { createLogger } from '@org/shared-observability';
import { WORKFLOW_NAME, type PropostaEvent, type StepResult } from './types.js';

const logger = createLogger({ service: WORKFLOW_NAME, handler: 'manual-review' });

/**
 * Applies a human reviewer's decision to a proposal that was flagged
 * IN_REVIEW. The decision is expected on `event.review`.
 */
export const handler = async (
  event: PropostaEvent,
  context: Context
): Promise<StepResult> => {
  const proposta = event.payload;
  const review = event.review ?? { approve: false, reviewer: 'unknown' };
  const status: StepResult['status'] = review.approve ? 'APPROVED' : 'REJECTED';

  logger.info('proposta.reviewed', {
    requestId: context.awsRequestId,
    propostaId: proposta.id,
    reviewer: review.reviewer,
    status,
  });

  return {
    workflow: WORKFLOW_NAME,
    step: 'manual-review',
    propostaId: proposta.id,
    status,
    detail: { reviewer: review.reviewer },
  };
};

File: apps/workflow-a/Handlers/ManualReviewHandler.cs
using Amazon.Lambda.Core;
using Acme.Shared.Domain;
using Acme.Shared.Observability;

namespace Acme.WorkflowA.Handlers;

/// <summary>
/// Applies a human reviewer's decision to a proposal that was flagged InReview.
/// </summary>
public sealed class ManualReviewHandler
{
    private static readonly IStructuredLogger Logger = StructuredLogger.Create(
        new Dictionary<string, object?> { ["service"] = WorkflowConstants.Name, ["handler"] = "manual-review" });

    public Task<StepResult> Handle(ManualReviewInput input, ILambdaContext context)
    {
        var proposta = input.Proposta;
        var status = input.Review.Approve ? Status.Approved : Status.Rejected;

        Logger.Info("proposta.reviewed", new Dictionary<string, object?>
        {
            ["requestId"] = context.AwsRequestId,
            ["propostaId"] = proposta.Id,
            ["reviewer"] = input.Review.Reviewer,
            ["status"] = status.ToString(),
        });

        return Task.FromResult(new StepResult(
            WorkflowConstants.Name, "manual-review", proposta.Id, status,
            new Dictionary<string, object?> { ["reviewer"] = input.Review.Reviewer }));
    }
}
